### PR TITLE
[pkg-config] Add new port

### PR DESCRIPTION
**Pull request by Neumann-A**
Just for the fun of it. This is a native windows build and does not require msys on windows after build. 

Probably requires the CI fixes for MSYS to build succesful?

### CODE REVIEW

File: ports/pkg-config/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+else()
+    set(GLIB_LIBS_DEBUG "-L${CURRENT_INSTALLED_DIR}/lib ${GLIBS} -lz -pthread")
+    set(GLIB_LIBS_RELEASE "-L${CURRENT_INSTALLED_DIR}/lib ${GLIBS} -lz -pthread")
+endif()
+
+set(ENV{GLIB_CFLAGS} "-I${CURRENT_INSTALLED_DIR}/include")
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    CONFIG_DEPENDENT_ENVIROMNENT "GLIB_LIBS"
```
Comment (c72578) [inline ports/pkg-config/portfile.cmake:26]: Please replace (8 occurrences):
ENVIROMNENT->ENVIRONMENT
enviromnent->environment

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 2]
```diff
@@ -44,6 +44,11 @@
 ## ### OPTIONS_DEBUG
 ## Additional options passed to configure during the Debug configuration. These are in addition to `OPTIONS`.
 ##
+## ### CONFIG_DEPENDENT_ENVIROMNENT
+## List of additional configuration dependent enviroment variables to set.
```
Comment (c72578) [inline scripts/cmake/vcpkg_configure_make.cmake:48]: enviroment->environment

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 3]
```diff
@@ -44,6 +44,11 @@
 ## ### OPTIONS_DEBUG
 ## Additional options passed to configure during the Debug configuration. These are in addition to `OPTIONS`.
 ##
+## ### CONFIG_DEPENDENT_ENVIROMNENT
+## List of additional configuration dependent enviroment variables to set. 
+## Pass SOMEVAR to set the enviroment and have SOMEVAR_(DEBUG|RELEASE) set in the portfile to the appriopiate values
```
Comment (c72578) [inline scripts/cmake/vcpkg_configure_make.cmake:49]: appriopiate->appropriate

**Comment by Neumann-A** [comment]
Hmm wondering if this pkg-config build on windows can actually handle nativ windows paths. The msys version always requires converted unix like paths in PKG_CONFIG_PATH and also returns unix like paths if queried for -L paths.

**Review by LilyWangL — APPROVED** [review]

**Comment by dan-shaw** [comment]
We have not really figured out how Vcpkg will acquire tools. Right now, the best way is to use `vcpkg_find_acquire_program`. I'm going to archive this PR for now, since we do not plan to build tools in Vcpkg yet. Please reopen if I'm missing something.

**Comment by Neumann-A** [comment]
hmm now i need to extract the changes to `vcpkg_configure_make.cmake` and put them into #11836. Hoped i could get away with them ;)

**Pull request closed without merging**